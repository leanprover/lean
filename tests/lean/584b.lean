section
  variable (A : Type)

  section
    variables (a b : A)
    variable  (H : a = b)

    definition tst₁ := a

    check @tst₁

    variable {A}

    definition tst₂ := a

    check @tst₂ -- A is implicit

    lemma symm₂ : b = a := eq.symm H

    check @symm₂
  end

  variable (a : A)
  definition tst₃ := a

  check @tst₃  -- A is explicit again
end
