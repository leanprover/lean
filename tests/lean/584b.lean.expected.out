tst₁ : Π (A : Type), A → A
tst₂ : Π {A : Type}, A → A
symm₂ : ∀ {A : Type} (a b : A), a = b → b = a
tst₃ : Π (A : Type), A → A
